### PR TITLE
Document task scheduling

### PR DESCRIPTION
**Pull request by settermjd**
## Description

Document task scheduling functionality as outlined in https://github.com/owncloud/ios-app/pull/484.

## Related Issue

- https://github.com/owncloud/ios-app/pull/484
- https://github.com/owncloud/docs/issues/1900

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by settermjd** [comment]
@mneuwert, I've attempted to take what you wrote and work it into the documentation. Can you review it and provide your feedback? Secondly, do you have more information on the bookmark update task?

**Comment by settermjd** [comment]
@mneuwert, if you're happy with the content of this PR, can you approve and merge it, please?

**Comment by settermjd** [comment]
@tboerger / @mneuwert can you help out with the failing bitrise check, pls?

**Comment by michaelstingl** [comment]
> @tboerger / @mneuwert can you help out with the failing bitrise check, pls?

@hosy could you have a look?

**Comment by mneuwert** [comment]
@michaelstingl timeout in bitrise when executing unit tests. May be temporary bitrise resource issue? Don't see anything suspicious in the logs.

**Comment by michaelstingl** [comment]
@settermjd add a new commit to trigger CI?

**Comment by settermjd** [comment]
> @settermjd add a new commit to trigger CI?

Done!

File: docs/modules/ROOT/pages/ios_task_scheduling.adoc [hunk 1]
```diff
@@ -0,0 +1,33 @@
+= Task Scheduling
+
+Background tasks are scheduled based on the app's current context, and that context is based on a combination of factors. 
+These factors include:
+
+* Is the app backgrounded or in the foreground?
+* Is WiFi available?
+* Is the device in Low Power Mode?
+* Is the device connected to external power?
+* Has the photo library has changed?
```
Comment (phil-davis) [inline docs/modules/ROOT/pages/ios_task_scheduling.adoc]:
```suggestion
* Has the photo library changed?
```

File: docs/modules/ROOT/pages/ios_task_scheduling.adoc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+== Tasks 
+
+Currently, two tasks are available:
+
+* *Instant photo uploads:* This task is triggered when:
+** The photo library has new assets
```
Comment (phil-davis) [inline docs/modules/ROOT/pages/ios_task_scheduling.adoc]:
```suggestion
** The photo library has new photos, and;
```
IMO "ordinary mortals" will not know that "assets" means photos.
IMO this task is triggered when A and B are both true - so we can be explicit about that

File: docs/modules/ROOT/pages/ios_task_scheduling.adoc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+* Is WiFi available?
+* Is the device in Low Power Mode?
+* Is the device connected to external power?
+* Has the photo library has changed?
+
+== Tasks 
+
+Currently, two tasks are available:
+
+* *Instant photo uploads:* This task is triggered when:
```
Comment (phil-davis) [inline docs/modules/ROOT/pages/ios_task_scheduling.adoc]: Is this "photo and video"?
If so, then "The photo library has new photos" will need enhancement also.

File: docs/modules/ROOT/pages/ios_task_scheduling.adoc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+** The photo library has new assets
+** The user is connected to a WiFi network
+* *Bookmark update task:* This task is triggered by the background fetch event scheduled by iOS
+
+=== Instant Photo Upload
+
+There are some things to be aware of, regarding Instant Photo Upload.
+
+* Photo and video upload can be enabled and disabled separately.
+* Before instant media upload can begin, the account and upload path have to be selected
```
Comment (phil-davis) [inline docs/modules/ROOT/pages/ios_task_scheduling.adoc]: why is it "instant media upload" here, but the heading... is "Instant Photo Upload"

File: docs/modules/ROOT/pages/ios_task_scheduling.adoc [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+* Changes in the photo library are detected when the app goes into the foreground.
```
Comment (phil-davis) [inline docs/modules/ROOT/pages/ios_task_scheduling.adoc]: How does this line interact with "This task is triggered when: The photo library has new assets"
This line makes me think that the task is not going to get triggered until there are new photos AND the app goes into the foreground.

File: docs/modules/ROOT/pages/ios_task_scheduling.adoc (hunk 3 above)
Comment (settermjd) [reply docs/modules/ROOT/pages/ios_task_scheduling.adoc]: Thanks for spotting that.

File: docs/modules/ROOT/pages/ios_task_scheduling.adoc (hunk 4 above)
Comment (settermjd) [reply docs/modules/ROOT/pages/ios_task_scheduling.adoc]: Good point. Well spotted.

File: docs/modules/ROOT/pages/ios_task_scheduling.adoc (hunk 5 above)
Comment (settermjd) [reply docs/modules/ROOT/pages/ios_task_scheduling.adoc]: Will review and update.

Comment (settermjd) [reply docs/modules/ROOT/pages/ios_task_scheduling.adoc]: @mneuwert, in #84, you wrote: 

> Changes in the photo library are detected when app goes into foreground (no signification location change detection magic as in some other apps to make it work in background). However, as soon as NSURLSession instances are created uploads will continue in the background.

Does this apply to videos as well?

Comment (michaelstingl) [reply docs/modules/ROOT/pages/ios_task_scheduling.adoc]: "Assets" in the "photo library" include videos as well.

Comment (settermjd) [reply docs/modules/ROOT/pages/ios_task_scheduling.adoc]: Thank you.

**Comment by hosy** [comment]
@mneuwert please can you have a look, if the documentation is still correct after your changes?

**Comment by mneuwert** [comment]
@hosy I think task scheduling is WIP at the moment and it would make sense to finish it first and then revise the documentation...

**Comment by mmattel** [comment]
Closing, already implemented in the new ios docs repo

**Pull request closed without merging**